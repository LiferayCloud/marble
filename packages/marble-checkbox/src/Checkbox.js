import Component from 'metal-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './Checkbox.soy.js';

/**
 * Checkbox component.
 */
class Checkbox extends Component {
  handleCheck(e) {
    this.emit('check', e);
  }
}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
Checkbox.STATE = {
  /**
   * The checked status of the input element.
   * @type {!Bool}
   */
  checked: Config.bool(),

  /**
   * The name and ID of the input element.
   * @type {!String}
   */
  id: Config.string(),

  /**
   * The label of the input element.
   * @type {!String}
   */
  label: Config.string(),

  /**
   * The value of the input element.
   * @type {!String}
   */
  value: Config.string()
};

Soy.register(Checkbox, templates);

export {Checkbox};
export default Checkbox;
