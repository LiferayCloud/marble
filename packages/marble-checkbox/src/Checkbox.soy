{namespace Checkbox}

/**
 * This renders the component's whole content.
 */
{template .render}
  {@param? id: string}
  {@param? value: string}
  {@param? label: string}
  {@param? checked: bool}

  {let $checkedAttr kind="attributes"}
    {if $checked}
      checked="checked"
    {/if}
  {/let}

  <div class="checkbox">
    <input id="{$id}" name="{$id}" value="{$value}" type="checkbox" {$checkedAttr}>
    <label for="{$id}">{$label}</label>
  </div>
{/template}
