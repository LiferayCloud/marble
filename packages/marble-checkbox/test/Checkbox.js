import Checkbox from '../src/Checkbox';

let checkbox;

describe('Checkbox', () => {
  afterEach(() => {
    if (checkbox) {
      checkbox.dispose();
    }
  });

  it('should generate the default markup', () => {
    checkbox = new Checkbox();

    expect(checkbox).toMatchSnapshot();
  });
});
