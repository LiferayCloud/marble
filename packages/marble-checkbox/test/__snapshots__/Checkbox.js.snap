// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should generate the default markup 1`] = `
<div class="checkbox">
  <input type="checkbox">
  <label></label>
</div>
`;
